packages/ra-core/src/controller/field/ReferenceManyFieldController.spec.tsx

```diff
@@ -7,6 +7,76 @@ import renderWithRedux from '../../util/renderWithRedux';
 
 describe('<ReferenceManyFieldController />', () => {
     it('should set loaded to false when related records are not yet fetched', async () => {
+        const ComponentToTest = ({ loaded }: { loaded?: boolean }) => {
+            return <div>loaded: {loaded.toString()}</div>;
+        };
+
+        const { queryByText } = renderWithRedux(
+            <ReferenceManyFieldController
+                resource="foo"
+                source="items"
+                reference="bar"
+                target="foo_id"
+                basePath=""
+            >
+                {props => <ComponentToTest {...props} />}
+            </ReferenceManyFieldController>,
+            {
+                admin: {
+                    references: {
+                        oneToMany: {},
+                        possibleValues: {},
+                    },
+                    resources: {
+                        bar: { data: {} },
+                        foo: { data: {} },
+                    },
+                },
+            }
+        );
+
+        expect(queryByText('loaded: false')).not.toBeNull();
+        await waitFor(() => {
+            expect(queryByText('loaded: true')).not.toBeNull();
+        });
+    });
+
+    it('should set loaded to true when related records have been fetched', async () => {
+        const ComponentToTest = ({ loaded }: { loaded?: boolean }) => {
+            return <div>loaded: {loaded.toString()}</div>;
+        };
+
+        const { queryByText } = renderWithRedux(
+            <ReferenceManyFieldController
+                resource="foo"
+                source="items"
+                reference="bar"
+                target="foo_id"
+                basePath=""
+            >
+                {props => <ComponentToTest {...props} />}
+            </ReferenceManyFieldController>,
+            {
+                admin: {
+                    references: {
+                        oneToMany: { ids: [] },
```

Comment:
same. Also, you test with a list of empty ids, you miss a test wit ha list of non-empty ids